portal/server/helpers/constructPayload.js

```diff
@@ -5,16 +5,24 @@
  * @param {Object} body Request body (req.body)
  * @param {Array} properties Interested properties to be added into payload
  * @param {Boolean} csrf Include CSRF token, defaulted to `true`
+ * @param {Boolean} propertyEmpty List of property names that can't have empty value as they will have object value later, defaulted to True
  * @returns {Object} Payload
  */
-const constructPayload = (body, properties, csrf = true) => {
+const constructPayload = (body, properties, propertyEmpty, csrf = true) => {
   let payload = {};
-
+  const bodyCopy = { ...body };
```

Comment:
Please rename `bodyCopy` to `payload`.

Reply:
set `canPropertyEmpty = false` and re-written comment

Reply:
Please rename bodyCopy to payload.

Reply:
FYI @abhi-markan we already have variable called `payload`, we construct it using list of `properties` so not all data from body is passed.

`bodyCopy` could be renamed to make more sense or code could be changed to avoid need of copying body.

code change 
```
properties
    .filter((property) => property in body)
    .filter((property) => canPropertyBeEmpty || body[[property]] !=='') 
    .forEach((property) => {
      payload = {
        ...payload,
        [property]: body[[property]],
      };
    });
```

Reply:
removed `bodyCopy` and added `.filter((property) => canPropertyBeEmpty || body[[property]] !== '')`